Sources/CryptoSwiftTests/AESTests.swift

```diff
@@ -5,16 +5,161 @@
 //  Created by Marcin Krzyzanowski on 27/12/14.
 //  Copyright (c) 2014 Marcin Krzyzanowski. All rights reserved.
 //
+
 import XCTest
-@testable import CryptoSwift
+import CryptoSwift
+import Foundation
 
 final class AESTests: XCTestCase {
+    
+    static let allTests: [(String, AESTests -> () throws -> Void)] = [("testAES_encrypt2", testAES_encrypt2), ("testAES_encrypt2", testAES_encrypt2), ("testAES_encrypt3", testAES_encrypt3), ("testAES_encrypt", testAES_encrypt), ("testAES_encrypt_cbc_no_padding", testAES_encrypt_cbc_no_padding), ("testAES_encrypt_cbc_with_padding", testAES_encrypt_cbc_with_padding), ("testAES_encrypt_cfb", testAES_encrypt_cfb), ("testAES_encrypt_cfb_long", testAES_encrypt_cfb_long), ("testAES_encrypt_ofb128", testAES_encrypt_ofb128), ("testAES_encrypt_ofb256", testAES_encrypt_ofb256), ("testAES_encrypt_pcbc256", testAES_encrypt_pcbc256), ("testAES_encrypt_ctr", testAES_encrypt_ctr), ("testAES_encrypt_ctr_irregular_length", testAES_encrypt_ctr_irregular_length), ("testAESWithWrongKey", testAESWithWrongKey), ("testAES_encrypt4", testAES_encrypt4)]
+    
     // 128 bit key
     let aesKey:[UInt8] = [0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0f]
+    
+    // MARK: - Functional Tests
 
-    func testAES_encrypt2() {
-        let key:[UInt8]   = [0x36, 0x37, 0x39, 0x66, 0x62, 0x31, 0x64, 0x64, 0x66, 0x37, 0x64, 0x38, 0x31, 0x62, 0x65, 0x65];
-        let iv:[UInt8]    = [0x6b, 0x64, 0x66, 0x36, 0x37, 0x33, 0x39, 0x38, 0x44, 0x46, 0x37, 0x33, 0x38, 0x33, 0x66, 0x64]
+    func testAES_encrypt4() {
```

Comment:
AES encrypting with 512 bit key crashes on Linux. testAES_encrypt4() is code from a project I'm working on.